src/main/scala/chisel3/compatibility.scala

```diff
@@ -59,16 +69,22 @@ package object Chisel {
       }
     }
   }
+
+  @nowarn("msg=Chisel compatibility mode is deprecated")
   implicit class cloneTypeable[T <: Data](target: T) {
     import chisel3.reflect.DataMirror
+    @deprecated("Chisel compatibility mode is deprecated. Use the chisel3 package instead.", "Chisel 3.6")
     def chiselCloneType: T = {
       DataMirror.internal.chiselTypeClone(target).asInstanceOf[T]
     }
   }
 
+  @deprecated("Chisel compatibility mode is deprecated. Use the chisel3 package instead.", "Chisel 3.6")
   type ChiselException = chisel3.internal.ChiselException
 
+  @deprecated("Chisel compatibility mode is deprecated. Use the chisel3 package instead.", "Chisel 3.6")
   type Data = chisel3.Data
+  @deprecated
```

Comment:
```suggestion
  @deprecated("Chisel compatibility mode is deprecated. Use the chisel3 package instead.", "Chisel 3.6")
```

Reply:
Without fixing this, we get 14 extra warnings in `CompatibilitySpec` (because the warning suppression doesn't match the default `@deprecated` message).